cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtilsTest.java

```diff
@@ -164,4 +167,52 @@ public void testAutoRebalanceStatusNewAddedNodes() {
         assertThat(kafkaStatus.getAutoRebalance().getModes().get(0).getMode(), is(KafkaAutoRebalanceMode.ADD_BROKERS));
         assertThat(kafkaStatus.getAutoRebalance().getModes().get(0).getBrokers().stream().collect(Collectors.toSet()).equals(Set.of(3, 4, 5, 6)), is(true));
     }
+
+    @Test
+    public void testUpdateStatusCondition() {
+        KafkaRebalanceStatus status = new KafkaRebalanceStatusBuilder().withConditions(new ArrayList<>()).build();
+
+        // Test Warning condition can be added when there are no existing conditions
+        KafkaRebalanceUtils.addWarningCondition(status, exception("Example error 0"));
+        assertThat(status.getConditions().size(), is(1));
+        assertThat(KafkaRebalanceUtils.getWarningCondition(status).getType(), is("Warning"));
+
+        // Test Warning condition addition doesn't remove existing conditions.
+        Condition c0 = new ConditionBuilder()
+                .withType("Rebalancing")
```

Comment:
Don't we have an enum for these Rebalancing conditions?

Reply:
Fixed.